### DISCUSSION TITLE
`response_model` not working for list[] responses

### DISCUSSION THREAD

**Discussion by ermal-abiti** [fastapi/fastapi#10040] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()

class Item(BaseModel):
    name: str
    price: float
    description: str


@app.post("/items/", response_model=list[Item], response_model_exclude={"price",})
async def create_item(item: Item):
    return [item]
```

### Description

After I call the `/items/` endpoint, i expect the following response as i wanted to exclude `price` from the model :
```json
[
  {
    "name": "string",
    "description": "string"
  }
]
```
but I am getting 
```json
[
  {
    "name": "string",
    "price": 0,
    "description": "string"
  }
]
```

Is there any solution for this ?

### Operating System

macOS

### FastAPI Version

0.101.0

### Pydantic Version

2.1.1

### Python Version

3.11

**Comment by tojimahammatov** [comment]
It worked with fastapi version 0.92.0, pydantic 1.10.5, and python 3.10, though I guess fastapi version matters. `response_model_exclude` parameter type changed to `IncEx` (complex Set and Dict types still though) in latest version via `fastapi.types` where 0.92.0 uses `Union[Set, Dict]` from `fastapi.encoders`. Maybe it can help for more debugging. I came up to this point only ). 

Most probably, I think we need to provide `{"price", }` value in a different way.

**Comment by JavierSanchezCastro** [comment, marked as answer]
Hi!

It's not possible to exclude when the response is a list, also I recommend data [filtering.](https://fastapi.tiangolo.com/tutorial/response-model/?h=filte#return-type-and-data-filtering)

See the [tip](https://fastapi.tiangolo.com/tutorial/response-model/#response_model_include-and-response_model_exclude)

This should work as expected:

```python3
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

class ItemBase(BaseModel):
    name: str
    description: str

class CreateItem(ItemBase):
    price: float

class ShowItem(ItemBase):
    pass


@app.post("/items/", response_model=list[ShowItem])
async def create_item(item: CreateItem):
    return [item]
```

**Comment by Akashh2151** [comment]
It seems like you are trying to exclude the "price" field from the response model when using FastAPI and Pydantic. However, in your current code, the "price" field is still included in the response.

To exclude a field from the response model, you can use the response_model_exclude parameter when defining the route. In your code, you are using it correctly, but there might be a version issue or a mistake in the usage.

In FastAPI 0.101.0, the response_model_exclude parameter should be a set of field names to exclude. However, it seems that it's not working as expected in your case.

Here are a few things you can try to resolve the issue:

Update FastAPI and Pydantic: Ensure that you are using the latest versions of FastAPI and Pydantic. Newer versions may have fixed this issue.

Use a List as a Response Model: Instead of using list[Item] as the response model, try using a List of dictionaries. You can manually create the response format you want.                                                                                                                                                                                            For example:                                                                                                                                                                                                 
 `from fastapi import FastAPI

app = FastAPI()

class Item(BaseModel):
    name: str
    price: float
    description: str

@app.post("/items/", response_model=list[dict], response_model_exclude={"price"})
async def create_item(item: Item):
    return [{"name": item.name, "description": item.description}]
`                                                                                                                                                                                                                                                                Check for Any Middlewares or Overrides: Make sure that there are no middlewares or custom response overrides in your FastAPI application that might be modifying the response.
If none of these solutions work, consider upgrading to a newer version of FastAPI, as newer versions often come with bug fixes and improvements. Additionally, check the FastAPI GitHub repository or discussion forums for any known issues related to response_model_exclude for your specific version.
